src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs

```diff
@@ -147,14 +147,19 @@ public static JsonDocument Parse(string json, JsonReaderOptions readerOptions =
             JsonReaderOptions readerOptions,
             byte[] extraRentedBytes)
         {
+            if (readerOptions.MaxDepth == 0)
+            {
+                readerOptions.MaxDepth = JsonReaderState.DefaultMaxDepth;
```

Comment:
If MaxDepth is moving to JsonReaderOptions, I'd expect DefaultMaxDepth to move as well; it's not just about reader state anymore.